crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -831,6 +832,27 @@ impl IndexerReader {
         })
     }
 
+    fn get_singleton_object(&self, struct_tag: StructTag) -> Result<Option<Object>, IndexerError> {
```

Comment:
If the provided struct tag is not of a singleton I guess it would be more safe to return an error instead of picking just the first object. But since the API is not public it should be fine for internal use as it is.

Reply:
It would be a responsibility of the caller to know that this type is a singleton in this case. As you said, it should be fine since it's an internal function.